docs/sources/tempo/traceql/construct-traceql-queries.md

```diff
@@ -0,0 +1,604 @@
+- `=~` (regular expression)
+- `!~` (negated regular expression)
+
+TraceQL uses Golang regular expressions.
+Online regular expression testing sites like https://regex101.com/ are convenient to validate regular expressions used in TraceQL queries.
+All regular expressions are treated as fully anchored.
+Regular expressions are anchored at both ends. This anchoring makes the queries faster and matches the behavior of PromQL, where regular expressions are also fully anchored.
+
+An unanchored query, such as: { span.foo =~ "bar" } is now treated as: { span.foo =~ "^bar$" }.
+
+If you use TraceQL with regular expressions in your Grafana dashboards and you want the unanchored behavior, update the queries to use the unanchored version, such as { span.foo =~ ".*bar.*"}.
+
+For example, to find all traces where an `http.status_code` attribute in a span are greater than `400` but less than equal to `500`:
+
+```
+{ span.http.status_code >= 400 && span.http.status_code < 500 }
+```
+
+This works for `http.status_code` values that are strings as well using lexographic ordering:
+
+```
+{ span.http.status_code >= "400" }
+```
+
+Find all traces where the `http.method` attribute is either `GET` or `DELETE`:
+
+```
+{ span.http.method =~ "DELETE|GET" }
+```
+
+Find all traces where `any_attribute` is not `nil` or where `any_attribute` exists in a span
+```
+{ span.any_attribute != nil }
+```
+
+### Field expressions
+
+Fields can also be combined in various ways to allow more flexible search criteria.
+A field expression is a composite of multiple fields that define all of the criteria that must be matched to return results.
+
+#### Examples
+
+Find traces with `success` `http.status_code` codes:
+
+```
+{ span.http.status_code >= 200 && span.http.status_code < 300 }
+```
+
+Find traces where a `DELETE` HTTP method was used and the intrinsic span status wasn't OK:
+
+```
+{ span.http.method = "DELETE" && status != ok }
+```
+
+Both expressions require all conditions to be true on the same span.
+The entire expression inside of a pair of `{}` must be evaluated as true on a single span for it to be included in the result set.
+
+In the above example, if a span includes an `.http.method` attribute set to `DELETE` where the span also includes a `status` attribute set to `ok`, the trace would not be included in the returned results.
+
+## Combine spansets using operators
+
+Spanset operators let you select different sets of spans from a trace and then make a determination between them.
+
+### Logical
+
+These spanset operators perform logical checks between the sets of spans.
+
+- `{condA} && {condB}` - The and operator (`&&`) checks that both conditions found matches.
+- `{condA} || {condB}` - The union operator (`||`) checks that either condition found matches. This functions as an "OR" statement.
+
+{{< admonition type="note" >}}
+A double pipe (`||`) is a union operator.
+A single pipe (`|`) lets you chain additional aggregators or conditions.
```

Comment:
```suggestion
A single pipe (`|`) pipes matched spans into another TraceQL selector, aggregate, or other function.
```

Reply:
I don't like adding `|` into this section b/c it's not a spanset operator.

Reply:
Okay. I'll remove this note.

Reply:
Done.